File: java/L1800/LeetCode1877.java
package L1800;

import java.util.Arrays;

public class LeetCode1877 {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int max = nums[0] + nums[nums.length - 1];
        for (int l = 0, r = nums.length - 1;l < r;++l, --r) {
            max = Math.max(max, nums[l] + nums[r]);
        }
        return max;
    }
}

File: cpp/L1800/LeetCode1877.cc
#include <algorithm>
#include <vector>
using std::vector;

class Solution {
public:
    int minPairSum(vector<int>& nums) {
        std::sort(nums.begin(), nums.end());
        int n = nums.size();
        int ans{0};
        for (int low{0}, high{n - 1};low < high;++low, --high) {
            ans = std::max(ans, nums[low] + nums[high]);
        }
        return ans;
    }
};
